client/client.go

```diff
@@ -316,11 +317,66 @@ func makeQueryString(data interface{}) (string, error) {
 	return "", fmt.Errorf("unsupported query string data: %#v", data)
 }
 
+// Remove all custom request serializer logic once APP-1331 is rolled out.
+type serializer func(any) ([]byte, error)
+
+func serializeQueryParamsToRequestBody(data any) ([]byte, error) {
+	m := make(map[string]any)
+	rv := reflect.ValueOf(data)
+	rt := reflect.TypeOf(data)
+	// If data is a map, just serialize it to JSON.
```

Comment:
Note that the Terraform provider depends on this behavior because it passes a map as the request body. This should be forward-compatible with TF migration to Go SDK for the Metastore/Catalog resources, because they will use the struct type and go in the following pathway.